sycl/include/CL/sycl/detail/kernel_program_cache.hpp

```diff
@@ -78,6 +82,7 @@ class KernelProgramCache {
 
   ProgramCacheT MCachedPrograms;
   KernelCacheT MKernelsPerProgramCache;
+  PlatformImplPtr MPlatform;
```

Comment:
Is it possible to have multiple plugins/platforms during the same runtime? Can we make  plugin a singleton?

Reply:
Currently no. But in the future yes. The design enables you to connect to multiple Plugins and have multiple platforms from different plugins at the same time. 
We have a GlobalPlugin which stores one plugin in use,it is currently used for OpenCL Interoperability constructors only. It will be a bad design to use it here, because we would like to delete that GlobalPlugin in the future.